src/app/AttributeCache.cpp

```diff
@@ -219,5 +219,75 @@ CHIP_ERROR AttributeCache::GetStatus(const ConcreteAttributePath & path, StatusI
     return CHIP_NO_ERROR;
 }
 
+struct compare
+{
+    bool operator()(const ConcreteDataAttributePathWithSize & x, const ConcreteDataAttributePathWithSize & y) const
+    {
+        return x.mSize < y.mSize;
+    }
+};
+
+uint32_t AttributeCache::OnUpdateDataVersionFilterList(DataVersionFilterIBs::Builder & aDataVersionFilterIBsBuilder,
+                                                       const Span<DataVersionFilter> & aDataVersionFilters)
+{
+    uint32_t number = 0;
+    CHIP_ERROR err  = CHIP_NO_ERROR;
+    TLV::TLVWriter backup;
+
+    std::set<ConcreteDataAttributePathWithSize, compare> clusterSet;
+    for (auto path = mChangedAttributeSet.begin(); path != mChangedAttributeSet.end(); path++)
+    {
+        ConcreteDataAttributePathWithSize cluster(*path);
+        UpdateClusterSize(cluster);
+        clusterSet.insert(cluster);
+    }
+
+    for (auto path = clusterSet.rbegin(); path != clusterSet.rend(); path++)
+    {
+        bool isIntersectDataVersionFilter = false;
+        DataVersionFilter dataVersionFilter(path->mEndpointId, path->mClusterId, path->mDataVersion);
+        aDataVersionFilterIBsBuilder.Checkpoint(backup);
+        VerifyOrExit(dataVersionFilter.IsValidDataVersionFilter(), err = CHIP_ERROR_INCORRECT_STATE);
+
+        // if the particular cached data version intersects with user provided data version filter, skip the cached one, and respect
+        // the user-provided one
+        for (auto & filter : aDataVersionFilters)
+        {
+            if (filter.IsSameCluster(*path))
+            {
+                isIntersectDataVersionFilter = true;
+                break;
+            }
+        }
+        if (isIntersectDataVersionFilter)
```

Comment:
This logic needs to pivot to intersection checks with the request paths instead.

Reply:
done, thanks